src/commands/ui.rs

```diff
@@ -1,11 +1,34 @@
+use std::fs;
+use std::io::Read;
+
+use ring::rand::SecureRandom;
+use ring::signature::KeyPair;
+use ring::{aead, agreement, digest, rand, signature};
+
+use crate::commands::ExitCode;
 use crate::options::Options;
+use crate::platform::data_dir;
+use crate::portable::local::instance_data_dir;
 use crate::print;
-use crate::commands::ExitCode;
 
 pub fn show_ui(options: &Options) -> anyhow::Result<()> {
     let connector = options.create_connector()?;
     let builder = connector.get()?;
-    let url = format!("http://{}:{}/ui", builder.get_host(), builder.get_port());
+    let mut url = format!("http://{}:{}/ui", builder.get_host(), builder.get_port());
+    if let Some(instance) = &options.conn_options.instance {
+        match if cfg!(windows) {
+            crate::portable::windows::get_ui_token(instance)
+        } else {
+            generate_jwt(instance)
+        } {
+            Ok(token) => {
+                url = format!("{}?authToken={}", url, token);
+            }
+            Err(e) => {
+                print::warn(format!("Cannot generate authToken: {:#}", e));
```

Comment:
Oh because here it still opens the UI in the browser, only without `authToken` set. The UI somehow may already have the token set in the browser cookie - like maybe it's just a linked remote instance I guess?